### PR TITLE
Dynamically routed schedules

### PR DESCRIPTION
**Pull request by quinnha**
Dynamically routed schedules via navbar, upcoming schedules, and scheduler page

https://github.com/LowerEarthOrbiters/Lower_Earth_Orbiters/assets/35006641/ea87c8a2-dcb0-4d5b-89fc-df15a56ca02e

### CODE REVIEW

File: src/leo-client-app/src/components/EditSchedules.tsx [hunk 1]
```diff
@@ -8,26 +8,38 @@ import "../styles.css";
...
+  };
   // const satelliteId = "655acd63d122507055d3d2ea";
   const satelliteId = router.query?.satelliteId?.toString() ?? "";
   // console.log(typeof(satelliteId), "hi");
   const adminUserId: string = "65a5e11fe0d601e0e8c4a385";
 
   // operator
   const userId: string = "65a8181f36ea10b4366e1dd9";
-  const scheduleId = "65a8182036ea10b4366e1de6";
   const isAdmin = adminUserId === userId;
```
Comment (RishiVaya) [inline src/leo-client-app/src/components/EditSchedules.tsx:27]: what about dynamic for user ids

File: src/leo-client-app/src/components/SatelliteName.tsx [hunk 2]
```diff
@@ -50,7 +18,7 @@ const SatelliteName = ({ noradId }: Props) => {
         marginTop: "50px",
       }}
     >
-      {satelliteName}
+      {name}
```
Comment (RishiVaya) [inline src/leo-client-app/src/components/SatelliteName.tsx:21]: satelliteName is a better name

File: src/leo-client-app/src/components/EditSchedules.tsx [hunk 3]
```diff
@@ -8,26 +8,38 @@ import "../styles.css";
...
-};
-
-const EditScheduler = ({ noradId }: Props) => {
+const EditScheduler = () => {
   const queryClient = useQueryClient();
   const router = useRouter();
+  const { satId, scheduleId } = router.query as {
+    satId: string;
+    scheduleId: string;
+  };
```
Comment (Umang-Rajkarnikar) [inline src/leo-client-app/src/components/EditSchedules.tsx:19]: can you instead do something like this

const satId = router.query.satId?.toString() ?? '';
const scheduleId = router.query. scheduleId?.toString() ?? '';

This way you won't have to cast a type

File: src/leo-client-app/src/components/Logs.tsx [hunk 4]
```diff
@@ -18,14 +18,22 @@ import "./styles/logs.css";
...
-  const satelliteId = "655acd63d122507055d3d2ea";
+  const router = useRouter();
+  let { satId } = router.query as {
+    satId: string;
+  };
+  if (!satId) {
+    satId = "655acd63d122507055d3d2ea";
+  }
+
+  const satelliteId = satId;
```
Comment (Umang-Rajkarnikar) [inline src/leo-client-app/src/components/Logs.tsx:36]: similar to above, but instead

const satId = router.query.satId?.toString() ?? "655acd63d122507055d3d2ea";

File: src/leo-client-app/src/pages/schedule-commands/[satId].tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import Navbar from "@/components/navbar/Navbar";
+import Scheduler from "@/components/Scheduler";
+import { Stack } from "@mui/material";
+import { useRouter } from "next/router";
+
+function SchedulerPage() {
+  const router = useRouter();
+  let { satId } = router.query as {
+    satId: string;
+  };
```
Comment (Umang-Rajkarnikar) [inline src/leo-client-app/src/pages/schedule-commands/[satId].tsx:12]: seems like this cna be removed ?

File: src/leo-client-app/src/components/EditSchedules.tsx [hunk 6]
```diff
@@ -8,26 +8,38 @@ import "../styles.css";
...
   // const satelliteId = "655acd63d122507055d3d2ea";
   const satelliteId = router.query?.satelliteId?.toString() ?? "";
   // console.log(typeof(satelliteId), "hi");
```
Comment (Umang-Rajkarnikar) [inline src/leo-client-app/src/components/EditSchedules.tsx:22]: remove

File: src/leo-client-app/src/components/EditSchedules.tsx [hunk 7]
```diff
@@ -8,26 +8,38 @@ import "../styles.css";
...
-
-const EditScheduler = ({ noradId }: Props) => {
+const EditScheduler = () => {
   const queryClient = useQueryClient();
   const router = useRouter();
+  const { satId, scheduleId } = router.query as {
+    satId: string;
+    scheduleId: string;
+  };
   // const satelliteId = "655acd63d122507055d3d2ea";
```
Comment (Umang-Rajkarnikar) [inline src/leo-client-app/src/components/EditSchedules.tsx:20]: remove

File: src/leo-client-app/src/components/EditSchedules.tsx (hunk 1 above)
Comment (quinnha) [reply src/leo-client-app/src/components/EditSchedules.tsx:27]: can be another pr, we only have 1 user for now anyways

File: src/leo-client-app/src/pages/schedule-commands/[satId].tsx (hunk 5 above)
Comment (quinnha) [reply src/leo-client-app/src/pages/schedule-commands/[satId].tsx:12]: think its good to leave in case we wanted to add more stuff to it

**Review by Umang-Rajkarnikar — APPROVED** [review]
lgtm

**Pull request merged**